www/cli/clearCache.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+// Copyright 2023 Catchpoint Systems Inc.
+// Use of this source code is governed by the Polyform Shield 1.0.0 license that can be
+// found in the LICENSE.md file.
+chdir('..');
+$MIN_DAYS = 2;
+
+require_once('common.inc');
+ignore_user_abort(true);
+set_time_limit(3300);   // only allow it to run for 55 minutes
+error_reporting(E_ALL);
+$is_cli = php_sapi_name() == "cli";
+if ($is_cli && function_exists('proc_nice')) {
+    proc_nice(19);
+}
+
+// bail if we are already running
+$lock = Lock("ClearLocalResults", false, 3600);
+if (!isset($lock)) {
+    if ($is_cli) {
+        echo "Clear local results process is already running\n";
+    }
+    exit(0);
+}
+
+$days_results_kept = null;
+if (GetSetting('days_results_kept')) {
+    $days_results_kept = GetSetting('days_results_kept');
+}
+
+$kept = 0;
+$deleted = 0;
+$log = fopen('./cli/clearedLocalResults.log', 'w');
+
+$MIN_DAYS = max($days_results_kept, 0.1);
+$UTC = new DateTimeZone('UTC');
+$now = time();
+
+// Delete the local results
+if (isset($days_results_kept)) {
+  $years = scandir('./results');
+  foreach ($years as $year) {
+      $yearDir = "./results/$year";
+      if (is_numeric($year) && is_dir($yearDir) && $year != '.' && $year != '..') {
+          $months = scandir($yearDir);
+          foreach ($months as $month) {
+              $monthDir = "$yearDir/$month";
+              if (is_dir($monthDir) && $month != '.' && $month != '..') {
+                  $days = scandir($monthDir);
+                  foreach ($days as $day) {
+                      $dayDir = "$monthDir/$day";
+                      if (is_dir($dayDir) && $day != '.' && $day != '..') {
+                          $elapsedDays = ElapsedDays($year, $month, $day);
+                          CheckDay($dayDir, "$year$month$day", $elapsedDays);
+                      }
+                  }
+                  @rmdir($monthDir);
+              }
+          }
+          @rmdir($yearDir);
+      }
+  }
+}
+
+if ($is_cli) {
+    echo "\nDone\n\n";
+}
+
+if ($log) {
+    fwrite($log, "Deleted: $deleted\nKept: $kept\n" . gmdate('r') . "\n");
+    fclose($log);
+}
+
+Unlock($lock);
+
+/**
+* Recursively check within a given day
+*
+* @param mixed $dir
+* @param mixed $baseID
+*/
+function CheckDay($dir, $baseID, $elapsedDays)
```

Comment:
Those "checks"  are a bit misleading 


```suggestion
function DeleteTestsExpiredOn($dir, $baseID, $elapsedDays)
```